### ISSUE TITLE
Migration file workflow when deploying

### ISSUE THREAD

**Issue by schickling** [graphcool/framework#627]
<a href="https://github.com/schickling"><img src="https://avatars2.githubusercontent.com/u/1567498?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [schickling](https://github.com/schickling)**
_Friday Sep 08, 2017 at 07:58 GMT_
_Originally opened as https://github.com/graphcool/cli/issues/173_

----

- [ ] Define migration filenames
- [ ] Define YAML DSL
- [ ] Define commands workflow for `deploy` and `diff`

**Comment by schickling** [comment]
<a href="https://github.com/timsuchanek"><img src="https://avatars1.githubusercontent.com/u/1094804?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [timsuchanek](https://github.com/timsuchanek)**
_Friday Sep 08, 2017 at 09:42 GMT_

----

# Proposal for Migration file workflow
## Migration Files
### Filenames
A new file called `migration.yml` will be created. The last migration will be renamed to `migration.prev.yml`, so the last migration is always available in `migration.prev.yml`.
Later we want to show migrations in the Console Dashboard, so users have an easier overview of which migrations have been executed.

### File Content
```yaml
migrationValues:
  - field: User.age
    value: "18"
  - field: Post.description
    value: "No description"

# use this if you don't want the data to be deleted
# rename:
#  - oldName: User.aage
#    newName: User.age
```

## Command Workflow
### deploy
When the deploy command is executed, the server tells the CLI if a migrationValue is required.
If that is the case, the CLI creates a template for the `migrationValues` and prompts the user to create migrationValues:
```sh
$ graphcool deploy
Deploy to project my-project...

Due to the changes in `types.graphql` you need to provide migration values.
A file called `migration.yml` has just been created with the following template:

migrationValues:
  - field: User.age
    value: ""

Please fill in a suiting migration value and run `graphcool deploy` again.
```

### diff
Same messages as in deploy:
```sh
$ graphcool deploy
Calculating diff of local and remote project definition...

Here are all the local changes:

  | (+)  A new type with the name `Post` is created.
  ├── (+)  A new field with the name `description` and type `String!` is created.

Due to the changes in `types.graphql` you need to provide migration values.
A file called `migration.yml` has just been created with the following template:

migrationValues:
  - field: User.age
    value: ""

Please fill in a suiting migration value to unblock the `graphcool deploy` command.
```

**Comment by marktani** [comment]
This proposal superseded by #1263
